bundles/org.openhab.binding.wundergroundupdatereceiver/src/main/java/org/openhab/binding/wundergroundupdatereceiver/internal/WundergroundUpdateReceiverServlet.java

```diff
@@ -0,0 +1,196 @@
+        synchronized (LOCK) {
+            try {
+                logger.debug("Starting servlet {} at {}", getClass().getSimpleName(), alias);
+                Dictionary<String, String> props = new Hashtable<>(1, 10);
+                httpService.registerServlet(alias, this, props, httpContext);
+                errorDetail = "";
+                active = true;
+            } catch (NamespaceException e) {
+                active = false;
+                errorDetail = "Servlet couldn't be registered - alias " + alias + " already in use";
+                logger.warn("Error during servlet registration - alias {} already in use", alias, e);
+            } catch (ServletException e) {
+                active = false;
+                errorDetail = "Servlet couldn't be registered - " + e.getMessage();
+                logger.warn("Error during servlet registration", e);
+            }
+        }
+    }
+
+    public void addHandler(WundergroundUpdateReceiverHandler handler) {
+        synchronized (this.handlers) {
+            if (this.handlers.containsKey(handler.getStationId())) {
+                throw new IllegalArgumentException(String
+                        .format("Handler handling request for stationId %s is already added", handler.getStationId()));
+            }
+            this.handlers.put(handler.getStationId(), handler);
+            if (!isActive()) {
+                activate();
+            }
+        }
+    }
+
+    public void removeHandler(String stationId) {
+        synchronized (this.handlers) {
+            WundergroundUpdateReceiverHandler handler = this.handlers.get(stationId);
+            if (handler != null) {
+                this.handlers.remove(stationId);
+            }
+            if (this.handlers.isEmpty()) {
+                deactivate();
+            }
+        }
+    }
+
+    @Override
+    protected void doGet(@Nullable HttpServletRequest req, @Nullable HttpServletResponse resp) throws IOException {
+        if (!active) {
+            return;
+        }
+        if (req == null) {
+            return;
+        }
+        if (resp == null) {
+            return;
+        }
+        if (req.getRequestURI() == null) {
+            return;
+        }
+        logger.trace("doGet {}", req.getQueryString());
+
+        Optional<WundergroundUpdateReceiverHandler> maybeHandler = Optional
+                .ofNullable(this.handlers.get(req.getParameter(STATION_ID_PARAMETER)));
+        maybeHandler.ifPresent(handler -> {
+            Map<String, String> states = req.getParameterMap().entrySet().stream()
+                    .collect(toMap(Map.Entry::getKey, e -> String.join("", e.getValue())));
+            String queryString = req.getQueryString();
+            if (queryString != null && queryString.length() > 0) {
+                states.put(LAST_QUERY, queryString);
+            }
+            handler.updateChannelStates(handler, states);
+        });
+
+        resp.setStatus(HttpServletResponse.SC_OK);
```

Comment:
Set current date in response. Weather station seems to use this time info to update it's local time.
```suggestion
        resp.setStatus(HttpServletResponse.SC_OK);
        resp.setDateHeader("Date", System.currentTimeMillis());
```

Reply:
```
22:09:44.171054 IP (tos 0x0, ttl 64, id 63989, offset 0, flags [DF], proto TCP (6), length 136)
    192-168-69-68.fritz.box.tproxy > wh2600.fritz.box.39076: Flags [P.], cksum 0x0c47 (incorrect -> 0x5ed6), seq 1:97, ack 512, win 30016, length 96
        0x0000:  4500 0088 f9f5 4000 4006 34ae c0a8 4544  E.....@.@.4...ED
        0x0010:  c0a8 4537 1f91 98a4 00ca 7ac5 0006 2420  ..E7......z...$.
        0x0020:  5018 7540 0c47 0000 4854 5450 2f31 2e31  P.u@.G..HTTP/1.1
        0x0030:  2032 3030 204f 4b0d 0a44 6174 653a 2053  .200.OK..Date:.S
        0x0040:  6174 2c20 3133 204d 6172 2032 3032 3120  at,.13.Mar.2021.
        0x0050:  3231 3a30 393a 3434 2047 4d54 0d0a 5365  21:09:44.GMT..Se
        0x0060:  7276 6572 3a20 4a65 7474 7928 392e 342e  rver:.Jetty(9.4.
        0x0070:  3230 2e76 3230 3139 3038 3133 290d 0a0d  20.v20190813)...
        0x0080:  0a73 7563 6365 7373                      .success
```

Reply:
Works for me now as follows using the code change from above:

1. Reboot of weather station
2. First GET request includes `dateutc=2012-08-22 2:14:45`
3. Servlet response includes `Date` Header
4. Actual date is used from now on ...

Reply:
The latest version resolves this finding 👍

Reply:
Could you please press the resolve conversation buttons of the resolved conversations? The thread is becoming unmanageably long.